pkg/storage/snapshot/restore.go

```diff
@@ -1007,6 +1143,31 @@ func (ctrl *VMRestoreController) deleteObsoleteDataVolumes(vmRestore *snapshotv1
 		}
 	}
 
+	// Garbage-collect original backend PVC if necessary
+	err := ctrl.deleteObsoleteBackendPVC(vmRestore, target)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ctrl *VMRestoreController) deleteObsoleteBackendPVC(vmRestore *snapshotv1.VirtualMachineRestore, target restoreTarget) error {
+	// Target should always exist at this point, just nil check for safety.
+	if target.Exists() && backendstorage.IsBackendStorageNeededForVM(target.VirtualMachine()) {
+		pvcs, err := ctrl.Client.CoreV1().PersistentVolumeClaims(vmRestore.Namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", restoreCleanupBackendPVCLabel, getCleanupLabelValue(vmRestore)),
+		})
+		if err != nil {
+			return err
+		}
+		for _, pvc := range pvcs.Items {
+			err = ctrl.Client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(context.Background(), pvc.Name, metav1.DeleteOptions{})
+			if err != nil {
```

Comment:
I wouldn't have returned an error immediately, but rather wrapped them into a single one and returned at the end of the loop. The main reason is because if one gives an error, you don't delete the following one.
Might be not necessary if the function is part of a reconciliation loop. 
We can do this in a follow-up PR